pkg/shp/cmd/build/run.go

```diff
@@ -92,11 +119,47 @@ func (r *RunCommand) tailLogs(pod *corev1.Pod) {
 	}
 }
 
+// onNoPodEventsYet reacts to the pod watcher telling us it has not received any pod events for our build run
+func (r *RunCommand) onNoPodEventsYet() {
+	r.Log(fmt.Sprintf("BuildRun %q log following has not observed any pod events yet.", r.buildRunName))
+	br, err := r.shpClientset.ShipwrightV1alpha1().BuildRuns(r.namespace).Get(r.cmd.Context(), r.buildRunName, metav1.GetOptions{})
+	if err != nil {
+		r.Log(fmt.Sprintf("error accessing BuildRun %q: %s", r.buildRunName, err.Error()))
+	} else {
+		c := br.Status.GetCondition(buildv1alpha1.Succeeded)
+		giveUp := false
+		msg := ""
+		switch {
+		case c != nil && c.Status == corev1.ConditionTrue:
+			giveUp = true
+			msg = fmt.Sprintf("BuildRun '%s' has been marked as successful.\n", br.Name)
+		case c != nil && c.Status == corev1.ConditionFalse:
+			giveUp = true
+			msg = fmt.Sprintf("BuildRun '%s' has been marked as failed.\n", br.Name)
+		case br.IsCanceled():
+			giveUp = true
+			msg = fmt.Sprintf("BuildRun '%s' has been canceled.\n", br.Name)
+		case br.DeletionTimestamp != nil:
+			giveUp = true
+			msg = fmt.Sprintf("BuildRun '%s' has been deleted.\n", br.Name)
+		case !br.HasStarted():
+			r.Log(fmt.Sprintf("BuildRun '%s' has been marked as failed.\n", br.Name))
+		}
+		if giveUp {
+			r.Log(msg)
+			r.Log(fmt.Sprintf("exting 'ship build run --follow' for BuildRun %q", br.Name))
+			r.stop()
+		}
+	}
```

Comment:
```suggestion
	if err != nil {
		r.Log(fmt.Sprintf("error accessing BuildRun %q: %s", r.buildRunName, err.Error()))
		return
	}
	c := br.Status.GetCondition(buildv1alpha1.Succeeded)
	switch {
	case c != nil && c.Status == corev1.ConditionTrue:
		r.Log(fmt.Sprintf("BuildRun '%s' has been marked as successful.\n", br.Name))
		r.stop()
	case c != nil && c.Status == corev1.ConditionFalse:
		r.Log(fmt.Sprintf("BuildRun '%s' has been marked as failed.\n", br.Name))
		r.stop()
	case br.IsCanceled():
		r.Log(fmt.Sprintf("BuildRun '%s' has been canceled.\n", br.Name))
		r.stop()
	case br.DeletionTimestamp != nil:
		r.Log(fmt.Sprintf("BuildRun '%s' has been deleted.\n", br.Name))
		r.stop()
	case !br.HasStarted():
		r.Log(fmt.Sprintf("BuildRun '%s' has been marked as failed.\n", br.Name))
	}
```

I think that this is cleaner and easier to tell exactly what is going on instead of having to track to the end of the switch statement to see what happens.

Reply:
I agree

*maybe* in one of my iterations prior to push I had something after the else block, but there's no telling :-)